File: c progamming/python/Sorting/insertion.cpp
#include <stdio.h>

void Insertion(int arr[], int n)
{
    int i, j;

    for (i = 1; i < n; i++) {
        while ( j >= 0;) {
            if (arr[j] > arr[i]) {
                int temp = arr[j];
                arr[j] = arr[i];
                arr[i] = temp;
            }
        }
    }
}

void printArr(int a[], int n) /* function to print the array */
{
    int i;
    for (i = 0; i < n; i++)
        printf("%d ", a[i]);
}

int main()
{
    int a[] = { 12, 31, 25, 8, 32, 17 };
    int n = sizeof(a) / sizeof(a[0]);
    printf("Before sorting array elements are - \n");
    printArr(a, n);
    Insertion(a, n);
    printf("\nAfter sorting array elements are - \n");
    printArr(a, n);
    return 0;
}    

File: java/Sorting/insertion.java
package Sorting;

import java.util.Arrays;

public class insertion {
    public static int[] insert(int[] arr,int n){
        for(int i=1;i<n;i++){
            int temp=arr[i];
            int j=i-1;
            while(j>=0 && arr[j]>temp){
                arr[j+1]=arr[j];
                j--;
            }
            arr[j+1]=temp;
        }
        return arr;
    }

    public static void main(String [] arg){
        int[] arr={23,24,11,77,10,7,24,33};
        int n= arr.length;
        System.out.print(Arrays.toString(insert(arr,n)));
    }
}
